srcjs/init_shiny.js

```diff
@@ -128,7 +136,13 @@ function initShiny() {
 
         var type = binding.getType(el);
         var effectiveId = type ? id + ":" + type : id;
-        currentValues[effectiveId] = binding.getValue(el);
+        inputItems[effectiveId] = {
```

Comment:
PR #1595 (commit 0ef15fa) removed the following code in `bindInputs` because we thought it was redundant:

```js
        if (shinyapp.isConnected()) {
          valueChangeCallback(binding, el, false);
        }
```

It turns out that the `false` in this call is important, and wasn't being used in the other call to `inputs.setInput` from `exports.bindAll`. So  we'll need to need to make sure that the value of `immediate` is set properly here.